valet/workfunc_test.go

```diff
@@ -127,3 +133,70 @@ func TestCalculateFileMD5(t *testing.T) {
 		assert.Equal(t, string(encoded), "5c9597f3c8245907ea71a89d9d39d08e")
 	}
 }
+
+func TestCompressFile(t *testing.T) {
+	tmpDir, err := ioutil.TempDir("", "TestCompressFile")
+	defer os.RemoveAll(tmpDir)
+	assert.NoError(t, err)
+
+	tmpFile := filepath.Join(tmpDir, "TestCompressFile")
+	of := fmt.Sprintf("of=%s", tmpFile)
+
+	// Makes a new random file each time
+	err = exec.Command("dd", "if=/dev/urandom", of,
+		"bs=1M", "count=50").Run()
+	assert.NoError(t, err)
+
+	randFile, err := NewFilePath(tmpFile)
+	assert.NoError(t, err)
+
+	// Calculates the MD5 of the uncompressed file
+	err = CreateMD5ChecksumFile(randFile)
+	assert.NoError(t, err)
+
+	checksumFile, err := NewFilePath(randFile.ChecksumFilename())
+	assert.NoError(t, err)
+
+	expectedMD5, err := ReadMD5ChecksumFile(checksumFile)
+	assert.NoError(t, err)
+
+	// Compress and check result
+	err = CompressFile(randFile)
+	assert.NoError(t, err)
+
+	compFile, err := NewFilePath(randFile.CompressedFilename())
+	assert.NoError(t, err)
+
+	err = compressedFileMatches(compFile.Location, expectedMD5)
+	assert.NoError(t, err)
+}
+
+func compressedFileMatches(path string, md5sum []byte) error {
+	f, err := os.Open(path)
+	defer f.Close()
```

Comment:
Check err first, because f may be nil on err

Reply:
Done.